auth_autologin_via_jwt_cookie/models/ir_http.py

```diff
@@ -0,0 +1,137 @@
+            if not email:
+                return super()._authenticate(endpoint)
+
+            user = (
+                request.env["res.users"]
+                .sudo()
+                .search(
+                    ["|", ("login", "=", email), ("email", "=", email)],
+                    limit=1,
+                )
+            )
+            if not user or not user.active:
+                return super()._authenticate(endpoint)
+
+            cls._force_login(user)
+        except Exception:
+            _logger.exception(
+                "JWT cookie autologin failed; falling back to default auth."
+            )
+            return super()._authenticate(endpoint)
+
+    @classmethod
+    def _get_jwks(cls, jwks_url: str) -> dict:
+        res = requests.get(jwks_url, timeout=5)
+        res.raise_for_status()
+
+        jwks = res.json()
+        return jwks
+
+    @classmethod
+    def _verify_jwt_with_jwks(cls, token, jwks_url):
+        """
+        Verify signature + standard claims (exp, nbf) using JWKS.
+        """
+        try:
+            headers = jwt.get_unverified_header(token)
+            kid = headers.get("kid")
+            if not kid:
+                _logger.warning("JWT missing kid header")
+                return None
+
+            jwks = cls._get_jwks(jwks_url)
+            keys = jwks.get("keys", [])
+            key_dict = next((k for k in keys if k.get("kid") == kid), None)
+            if not key_dict:
+                _logger.warning("No matching JWKS key for kid=%s", kid)
+                return None
+
+            public_key = jwk.construct(key_dict)
+
+            message, encoded_sig = token.rsplit(".", 1)
+            decoded_sig = base64url_decode(encoded_sig.encode("utf-8"))
+            if not public_key.verify(message.encode("utf-8"), decoded_sig):
+                _logger.warning("JWT signature verification failed")
+                return None
+
+            claims = jwt.get_unverified_claims(token)
+
+            exp = claims.get("exp")
+            if exp and time.time() > float(exp):
+                _logger.warning("JWT expired")
+                return None
+
+            return claims
+
+        except Exception:
+            _logger.exception("JWT verification error")
+            return None
+
+    @classmethod
+    def _get_email_from_userinfo(cls, userinfo_url: str, token: str) -> str | None:
+        try:
+            res = requests.get(
+                userinfo_url,
+                headers={"Authorization": f"Bearer {token}"},
+                timeout=5,
+            )
+            res.raise_for_status()
+            data = res.json()
+
+            email = (data.get("email") or "").strip()
+            return email or None
+
+        except Exception:
+            _logger.exception("Userinfo fetch failed")
+            return None
+
+    @classmethod
+    def _force_login(cls, user):
+        request.update_env(user=user.id)
```

Comment:
probably not enough to make the session persistent
I see it `auth_autologin_via_jwt_cookie.models.ir_http: Auto-authenticated user` multiple times in logs, only one time should happen